json_feature.cpp

```diff
@@ -1,27 +1,38 @@
 
 #include "json_feature.hpp"
 
-void JSONFeature::add_properties(const osmium::OSMObject& object, const char* id_name) {
+void JSONFeature::add_properties(const osmium::OSMObject& object) {
     m_writer.String("properties");
 
     m_writer.StartObject();
 
-    m_writer.String(id_name);
-    m_writer.Int64(object.id());
+    m_writer.String((m_attr_prefix + "id").c_str());
+    m_writer.Int64(object.type() == osmium::item_type::area ? osmium::area_id_to_object_id(object.id()) : object.id());
+
+    m_writer.String((m_attr_prefix + "type").c_str());
+    if (object.type() == osmium::item_type::area) {
+        if ((object.positive_id() & 0x1) == 0) {
```

Comment:
Ran into a problem with the `static_cast`: 

```
Users/r/gh/mapbox/minjur/json_feature.cpp:14:13: error: no matching conversion for static_cast from 'const osmium::OSMObject' to 'osmium::Area'
        if (static_cast<osmium::Area>(object).from_way()) {
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/r/gh/osmcode/libosmium/include/osmium/osm/area.hpp:115:11: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from
      'const osmium::OSMObject' to 'const osmium::Area' for 1st argument
    class Area : public OSMObject {
          ^
/Users/r/gh/osmcode/libosmium/include/osmium/osm/area.hpp:119:9: note: candidate constructor not viable: requires 0 arguments, but 1 was provided
        Area() :
        ^
```

Also no luck for a C-style cast.